pkg/costmodel/clusters/clustermap.go

```diff
@@ -210,22 +209,22 @@ func (pcm *PrometheusClusterMap) loadClusters() (map[string]*ClusterInfo, error)
 	}
 
 	// populate the local cluster if it doesn't exist
-	localID := env.GetClusterID()
-	if _, ok := clusters[localID]; !ok {
-		localInfo, err := pcm.getLocalClusterInfo()
-		if err != nil {
-			log.Warningf("Failed to load local cluster info: %s", err)
-		} else {
-			clusters[localInfo.ID] = localInfo
-		}
+	localInfo, err := pcm.getLocalClusterInfo()
```

Comment:
It depends on the implementation, which there is now one that pulls ClusterInfo the way it does in development, and one that now pulls ClusterInfo from the configured provider. If we pulled the cluster id from the environment variable on the hosted cluster, then it would be using the local cluster id and not the cluster id specific to the remote cluster. If we're running on the on-prem version, then the cluster id on ClusterInfo is pulled from the local provider: You can reference any of the `Provider` implementations for `ClusterInfo() (map[string]string, error)`. It looks like they use `env.GetClusterID()`: https://github.com/kubecost/cost-model/blob/c97efb88478965a08705be650bb4d9b6e08baa3b/pkg/cloud/gcpprovider.go#L537

Using the env var here was lazy, so this is an improvement.